common/main/root.xml

```diff
@@ -979,7 +979,9 @@ Warnings: All cp values have U+FE0F characters removed. See /annotationsDerived/
 						<dateFormatItem id="Ehms">E h:mm:ss a</dateFormatItem>
 						<dateFormatItem id="EHms">E HH:mm:ss</dateFormatItem>
 						<dateFormatItem id="Gy">G y</dateFormatItem>
+						<dateFormatItem id="GyM">GGGGG y-MM</dateFormatItem>
```

Comment:
The root and english locales should use G consistently instead of sometimes G and sometimes GGGGG (that is, default to the "medium" format instead of a mix of medium and narrow formats). This would then be overridden in the various languages.

Reply:
Actually, the procedure that we usually try to follow is to have GGGGG if there is a numeric month and G otherwise. I will convert accordingly (in the first version of this I just followed whatever convention was already used for GyMd, but that did not necessarily accord with the above)

Reply:
Well, let me amend that, looking at the actual eras. `root` only has abbreviated eras so I think only G should be used formats. For `en` we did want to use GGGGG for numeric formats (with numeric month) in the japanese calendar so we just have the single-letter era name. For the other calendars in `en` I agree we should use G.

Reply:
@macchiati OK, looking at the eras: in `root` we only want G (we only have abbreviated eras). In `en` we want G in most cases, certainly gregorian; however in the japanese calendar, for formats with numeric month we definitely want GGGGG to get the one-letter abbreviated era names. Made those changes, please take a look (and if you approve, please also merge if tests have passed).